app/scripts/views/choose_what_to_sync.js

```diff
@@ -0,0 +1,107 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+define([
+  'underscore',
+  'views/form',
+  'stache!templates/choose_what_to_sync',
+  'lib/promise'
+],
+function (_, FormView, Template, p) {
+  'use strict';
+
+  var View = FormView.extend({
+    template: Template,
+    className: 'choose-what-to-sync',
+
+    initialize: function () {
+      // Account data is passed in from sign up flow.
+      var data = this.ephemeralData();
+      if (! data || ! data.account || ! data.account.get('email')) {
+        this.navigate('signup');
+      }
+
+      this._account = data && this.user.initAccount(data.account);
+    },
+
+    getAccount: function () {
+      return this._account;
+    },
+
+    context: function () {
+      var account = this.getAccount();
+
+      return {
+        email: account.get('email')
+      };
+    },
+
+    submit: function () {
+      var self = this;
+      var account = self.getAccount();
+      var declinedEngines = self._getDeclinedEngines();
+      this._trackUncheckedEngines(declinedEngines);
+      account.set('declinedSyncEngines', declinedEngines);
```

Comment:
nit: the "engines" naming here is highly specific implementation terminology for sync.  I wonder if it'd be clearer to call them "datatypes" or something else in code on our side, since most readers of this code won't have much idea what an "engine" is in sync.  Then again, perhaps it's worth a little bit og "huh?" for new readers in order to have the most accurate name.